Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/storage/RecoveryCertificateStorage.kt

```diff
@@ -0,0 +1,63 @@
+package de.rki.coronawarnapp.covidcertificate.recovery.core.storage
+
+import android.content.Context
+import androidx.core.content.edit
+import com.google.gson.Gson
+import de.rki.coronawarnapp.util.di.AppContext
+import de.rki.coronawarnapp.util.serialization.BaseGson
+import de.rki.coronawarnapp.util.serialization.fromJson
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RecoveryCertificateStorage @Inject constructor(
+    @AppContext val context: Context,
+    @BaseGson val baseGson: Gson,
+) {
+
+    private val prefs by lazy {
+        context.getSharedPreferences("recovery_localdata", Context.MODE_PRIVATE)
+    }
+
+    private val gson by lazy {
+        baseGson.newBuilder().create()
+    }
```

Comment:
why do we need this? can't we just use `baseGson` directly ?

Reply:
thank you!